### ISSUE TITLE
mso_site - import bug or new resource required depending on maintainer intentions

### ISSUE THREAD

**Issue by mike-guy** [CiscoDevNet/terraform-provider-mso#121]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

terraform --version
Terraform v1.1.4
on windows_amd64

provider registry.terraform.io/ciscodevnet/mso v0.5.0

### MSO version

Nexus Dashboard
Version 2.1(2d)

Orchestrator
Version: 3.4(1i)

### APIC version and APIC Platform for Site Level Resources

Nexus Dashboard
Version 2.0(2g)

Multi-Site Orchestrator
Version: 3.3(1e)

Cisco dCloud lab "Cisco Nexus Dashboard Orchestrator for ACI Lab v1"

### Affected Resource(s)

* mso_site

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    mso = {
      source  = "CiscoDevNet/mso"
      version = "0.5.0"
    }
  }
  required_version = "~> 1.1.0"
}

provider "mso" {
  # creds as env vars
  url      = "https://198.18.133.100"
  platform = "nd"
}

resource "mso_site" "site" {}
```

### Expected Behavior
It is expected that an MSO resource and the attributes defined in the schema can be imported.

### Actual Behavior
Only a portion of attributes are expected.

This is either because a new resource type is required (e.g. mso_nd_site or similiar) or because of missing code. More details in the following sections.

### Question to maintainers
I'm happy to contribute code to help fix this, I'd just like to understand how you'd like to proceed.

1. New resource e.g. "mso_nd_site" - Based on what I'm seeing from the responses, this is likely what may be needed. The attributes for "mso_site" do not match up with the data you get back from /mso/api/v2/sites/<site_id>
2. Single resource with the ability to handle both response times and ensure all relevant attributes are imported - currently they are not.

More details in the following section.

### Steps to Reproduce

1. Create a site in an ND hosted version of MSO
2. Import the resource
3. Note that many of the attributes required by the resource are not imported:
    * username
    * password
    * login_domain
    * maintenance_mode
    * urls
    * location

#### ND (v2) vs MSO (v1)
The documentation on the nd settings isn't overly clear (nor am I particularly an ACI/MSO guy!) but I interpret it that if it's an ND hosted MSO instance, you'll need to use the "nd" platform setting in the provider. This doesn't appear to apply to all resource types however...

Looking at the code for mso_site:

* If the mso setting is configured it uses /api/v1/sites/<site_id>
* If the nd setting is configured it uses /mso/api/v2/sites/<site_id>

When the "nd" version is used, a number of schema attributes are not imported. However, the only one that COULD be imported from the data is urls - the others are not present.

#### Confusion
With the dCloud instance of MSO (on nexus dashboard) that I'm using, both the v1 and v2 URLs return data when accessed in a browser. I've not dug into the provider code, but there is obviously something somewhere that breaks if v1 is used for ND hence the implementation of the nd attribute.

##### v1 response (mso)
As you can see, the data matches that of the schema defined for "resource mso_site"
```json
{
    "id": "622b1412db71b3cb663aecfb",
    "name": "San Fran",
    "urls": [
        "https://apic1-a.dcloud.cisco.com"
    ],
    "username": "Cisco_MSO_QrRDjskP",
    "password": "******",
    "apicSiteId": "1",
    "labels": null,
    "platform": "on-premise",
    "apicVersion": "5.1(1h)",
    "maintenanceMode": false,
    "health": {}
}
```

##### v2 response (nd)
```json
{
    "id": "622b1412db71b3cb663aecfb",
    "common": {
        "siteId": "1",
        "name": "San Fran",
        "urls": [
            "https://apic1-a.dcloud.cisco.com:443"
        ],
        "crossUrls": [
            "https://198.18.133.201:443",
            "https://198.18.133.202:443",
            "https://198.18.133.200:443"
        ],
        "username": "Cisco_MSO_QrRDjskP",
        "platformType": "Apic",
        "siteGroup": "60f530940a99c901ec58d5ec",
        "siteVersion": "5.1(1h)"
    },
    "health": {},
    "apic": {},
    "dcnm": {}
}
```

**Comment by lhercot** [comment]
v1 APIs are still there in some versions of MSO on ND but are being deprecated and have been removed from some later version. As the v2 APIs are already present on earlier version we made the choice to switch to them. We might have missed some resources that are not yet using those v2 APIs.

I would like to keep a single resources that do both if possible. I am leaning to map the values that we can map and add the new model to the existing model. This should keep backward compatibility and in the future we can deprecate slowly.
